packages/cli/generators/relation/index.js

```diff
@@ -267,24 +314,69 @@ module.exports = class RelationGenerator extends ArtifactGenerator {
       );
     }
 
-    if (
-      this.artifactInfo.relationType === relationUtils.relationType.belongsTo
-    ) {
-      return;
+    this.artifactInfo.destinationModelPrimaryKey = await relationUtils.getModelPrimaryKeyProperty(
+      this.fs,
+      this.artifactInfo.modelDir,
+      this.artifactInfo.destinationModel,
+    );
+    if (this.artifactInfo.destinationModelPrimaryKey == null) {
+      return this.exit(
+        new Error(ERROR_DESTINATION_MODEL_PRIMARY_KEY_DOES_NOT_EXIST),
+      );
     }
 
+    if (this.artifactInfo.destinationModelPrimaryKey) {
+      this.artifactInfo.destinationModelPrimaryKeyType = relationUtils.getModelPropertyType(
+        this.artifactInfo.modelDir,
+        this.artifactInfo.destinationModel,
+        this.artifactInfo.destinationModelPrimaryKey,
+      );
+    }
+
+    // for controller usage;
     this.artifactInfo.targetModelPrimaryKey = await relationUtils.getModelPrimaryKeyProperty(
       this.fs,
       this.artifactInfo.modelDir,
       this.artifactInfo.destinationModel,
     );
 
+    if (
+      this.artifactInfo.relationType === relationUtils.relationType.belongsTo
+    ) {
+      // for belongsTo the fk is the id property of source model by default
+      // should check the existance of fk here.
+      // default source key for belongsTo
+      this.artifactInfo.defaultSourceKeyName =
+        utils.camelCase(this.artifactInfo.destinationModel) +
+        utils.toClassName(this.artifactInfo.destinationModelPrimaryKey);
+
+      return this.prompt([
+        {
+          type: 'string',
+          name: 'sourceKeyName',
```

Comment:
use `sourceKeyName`, `relationName`, and `defaultRelationName` to check if the source key of belongsTo is customized. These three properties also check if the `name` field of the relation metadata needs to be specified.